### PR TITLE
Add Marketplace documentation

### PR DESCRIPTION
**Pull request by dennisameling**
Belongs to https://github.com/mautic/mautic/pull/10518

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/10"></a>

### CODE REVIEW

File: docs/marketplace/listing.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            "mautic/core-lib": "^4.0"
+        }
+    }
+
+In addition to the information you already have in the composer.json for your plugin, please add:
+
+- ``type`` must have value ``mautic-plugin``. The Marketplace is filtering PHP packages by this tag. It is required to show up in the Marketplace.
+- ``extra.install-directory-name`` specifies the directory name for the bundle. Correct directory name is important for PSR4 autoloading. It must be the same as in the namespace in your PHP classes.
+- ``require.php`` be sure to specify the PHP version range that you test the plugin against. Ideally it should be the same as the Mautic's `supported PHP versions <https://www.mautic.org/download/requirements>`_. Do not allow users to install the plugin on versions that are not supported.
+- ``require.ext-*`` if your plugin require some PHP extension, please list them in the require section too. No one knows on what servers your plugin will run on.
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:46]:
```suggestion
- ``require.ext-*`` if your Plugin require some PHP extension, please list them in the require section too. As servers can have a wide variety of extensions enabled, this ensures that the correct extensions are available.
```

File: docs/marketplace/listing.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- ``require.mautic/core-lib`` it's important to specify which Mautic versions your plugin supports. Include only the versions you or your community have tested and confirmed the plugin works with.
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:47]:
```suggestion
- ``require.mautic/core-lib`` it's important to specify which Mautic versions your Plugin supports. Include only the versions you or your community have tested and confirmed the Plugin works with.
```

File: docs/marketplace/listing.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+Step 2: check best practices
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:49]:
```suggestion
Step 2: verify best practices
```

File: docs/marketplace/listing.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+----------------------------
+
+There's a few things we highly recommend when creating plugins for Mautic. Please refer to the :doc:`./best_practices` page for more details.
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:52]:
```suggestion
There's a few things which are highly recommend when creating Plugins for Mautic. Please refer to the :doc:`./best_practices` page for more details.
```

File: docs/marketplace/listing.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+Step 3: publish to Composer
+---------------------------
+
+When the composer.json is ready, follow the `Publishing Packages <https://packagist.org>`_ section directly in Packagist.
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:57]: Please use the external linking extension here!

File: docs/marketplace/listing.rst [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+Step 4: apply for the allowlist
+-------------------------------
+
+While the Mautic Marketplace is still in beta, you'll have to apply for our allowlist in order to show up in the Marketplace. We expect to move to stable with Mautic 4.2 in February 2022, after which the allowlist likely won't be needed anymore. Please refer to the :doc:`./allowlist_what_and_why` page for more details.
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:62]:
```suggestion
While the Mautic Marketplace is still in beta, you'll have to apply for our allowlist in order to show up in the Marketplace. It is expected that the Marketplace will move from beta to stable release with Mautic 4.2 in February 2022, after which the allowlist may be deprecated in favour of a blocklist. Please refer to the :doc:`./allowlist_what_and_why` page for more details.
```

File: docs/marketplace/listing.rst [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+Updating existing plugins for usage with the Marketplace
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:64]:
```suggestion
Updating existing Plugins for usage with the Marketplace
```

File: docs/marketplace/listing.rst [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+========================================================
+
+If you required ``mautic/composer-plugin`` in your plugin's dependencies in the past, please remove it. Support for Mautic plugins is now built into Composer, so you only have to set the type to ``mautic-plugin`` and Composer will automatically install your plugin into the ``plugins`` folder.
+
+Next to that, if you built your plugin for Mautic 3 originally, please check our `UPGRADE-4.0.md guide <https://github.com/mautic/mautic/blob/4.x/UPGRADE-4.0.md>`_ for the breaking changes we made in Mautic 4.
```
Comment (RCheesley) [inline docs/marketplace/listing.rst:69]:
```suggestion
Next to that, if you built your plugin for Mautic 3 originally, please check the `UPGRADE-4.0.md guide <https://github.com/mautic/mautic/blob/4.x/UPGRADE-4.0.md>`_ for the breaking changes in Mautic 4.
```

**Comment by RCheesley** [comment]
@dennisameling should also mention that Vale seems to be passing despite an awful lot of warnings - is it configured correctly?

**Comment by dennisameling** [comment]
@RCheesley 
- Added a build + checklinks step to CI - this ensures that external links can actually be reached (404/500/etc. error codes will fail the job)
- Added Vale + several plugins like GitLens to Gitpod, so that contributors see the warnings/suggestions right within their editor while working on docs
- Moved external links to their own files as requested
- Fixed several warnings & suggestions

There's two small TODOs left in this PR - adding screenshots and linking to the plugin creation docs (which I'll create in a separate PR)

File: docs/marketplace/allowlist_what_and_why.rst [hunk 9]
```diff
@@ -0,0 +1,27 @@
+============================================
+Allowlist: what's it and why is it needed?
```
Comment (RCheesley) [inline docs/marketplace/allowlist_what_and_why.rst:2]:
```suggestion
Allowlist: what is it and why is it needed?
```
Guessing this was a Vale correction but actually makes better grammatical sense to not abbreviate in that way.

**Review by RCheesley — CHANGES_REQUESTED** [review]
One small fix to grammar otherwise good to go. Let's not forget the TODOs!

Also as a side note, when I spun up Gitpod it asked me to install Python (the extension maybe?) and it says Vale is not available. Maybe something is not quite right there?

**Pull request merged**